# <center> Линейный оператор

In [2]:
import numpy as np

In [3]:
# 2.2
F = np.array([
    [1, 0, 0],
    [0, 0, 1],
    [0, -1, 0]
])
v = np.array([1, 2, 3])
print('Образ вектора v:', F @ v)

Образ вектора v: [ 1  3 -2]


In [4]:
u = np.array([1, -1])
v = np.array([1, 1])
F = np.array([
    [2, 1],
    [1, 2]
])
print('Образ u:', F @ u)
print('Образ v:', F @ v)

Образ u: [ 1 -1]
Образ v: [3 3]


# <center> Собственные числа и векторы

In [5]:
A = np.array([
    [3, 1],
    [2, 2]
]).T
# вычисляем собственные числа и собственные векторы
eig_values, eig_vectors = np.linalg.eig(A)
print('Собственные числа: \n', eig_values)
print('Собственные векторы: \n', eig_vectors)

Собственные числа: 
 [4. 1.]
Собственные векторы: 
 [[ 0.89442719 -0.70710678]
 [ 0.4472136   0.70710678]]


>Каждый собственный вектор соответствует своему собственному числу. То есть вектор $\vec{v_1}=(-0.894, 0.447)^T$ соответствует собственному числу $\lambda_1=4$, а собственный вектор $\vec{v_2}=(-0.707, 0.707)^T$ соответствует числу $\lambda_2=1$. 

In [6]:
# 4.1
A = np.array([
    [1, 3, 5],
    [4, -4, 9],
    [13, 7, 12]
])
eig_values, eig_vectors = np.linalg.eig(A)
print('Собственные числа: \n', eig_values.round())


Собственные числа: 
 [20. -4. -7.]


In [7]:
# 4.2
A = np.array([
    [1, 9, 4],
    [9, 4, 7],
    [4, 7, 12]
])
eig_values, eig_vectors = np.linalg.eig(A.T @ A)
print('Собственные числа: \n', eig_values.round())

Собственные числа: 
 [391.  46.  16.]
